yamllint/rules/comments.py

```diff
@@ -97,7 +97,9 @@ def check(conf, comment):
                     comment.column_no == 1 and
                     re.match(r'^!\S', comment.buffer[text_start:])):
                 return
-            elif comment.buffer[text_start] not in (' ', '\n', '\0'):
+            # We can test for both \r and \r\n just by checking first char
+            # \r itself is a valid newline on some older OS.
+            elif comment.buffer[text_start] not in {' ', '\n', '\r', '\x00'}:
```

Comment:
I'm not sure the O(1) search for 4 one-byte elements compensate the overhead of building a set + hashings required, but I admit it's nicer-looking now :wink:

Reply:
You might very well be right on that one 😏